src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -3134,6 +3140,120 @@ ElectricStorage::ElectricStorage( // main constructor
+                    ShowContinueError(state, format("{} = {:.3R}", DataIPShortCuts::cNumericFieldNames(i), DataIPShortCuts::rNumericArgs(i)));
+                }
+                errorsFound = true;
+            }
+            liIon_Vnom_default_ = DataIPShortCuts::lNumericFieldBlanks(13) ? 3.342 : DataIPShortCuts::rNumericArgs(13);
+            liIon_Qfull_ = DataIPShortCuts::lNumericFieldBlanks(14) ? 3.2 : DataIPShortCuts::rNumericArgs(14);
+            liIon_Qexp_ = DataIPShortCuts::lNumericFieldBlanks(15) ? 0.8075 * liIon_Qfull_ : DataIPShortCuts::rNumericArgs(15) * liIon_Qfull_;
+            liIon_Qnom_ = DataIPShortCuts::lNumericFieldBlanks(16) ? 0.976875 * liIon_Qfull_ : DataIPShortCuts::rNumericArgs(16) * liIon_Qfull_;
+            if (liIon_Qexp_ >= liIon_Qnom_) {
+                ShowSevereError(state, routineName + DataIPShortCuts::cCurrentModuleObject + "=\"" + DataIPShortCuts::cAlphaArgs(1) + "\", invalid entry.");
+                ShowContinueError(state, DataIPShortCuts::cNumericFieldNames(16) + " must be greater than " + DataIPShortCuts::cNumericFieldNames(15) + ".");
+                for (int i = 15; i <= 16; ++i) {
+                    ShowContinueError(state, format("{} = {:.3R}", DataIPShortCuts::cNumericFieldNames(i), DataIPShortCuts::rNumericArgs(i)));
+                }
+                errorsFound = true;
+            }
+            liIon_C_rate_ = DataIPShortCuts::lNumericFieldBlanks(17) ? 1.0 : DataIPShortCuts::rNumericArgs(17);
+            internalR_ = DataIPShortCuts::lNumericFieldBlanks(18) ? 0.09 : DataIPShortCuts::rNumericArgs(18);
+
+            maxAhCapacity_ = liIon_Qfull_ * parallelNum_;
+
+            if (!errorsFound) {
+                // Set the Lifetime model in SSC
+                // I'm using a raw pointer here because the the battery_t constructor expects it.
+                // The pointer is then passed into the battery_t where it is converted into a unique_ptr and persists along with that object.
+                // Therefore I am not deleting this pointer here because that will be handled by the battery_t class.
+                lifetime_t* battLifetime;
+                if (lifeCalculation_ == BatteryDegradationModelType::lifeCalculationYes) {
+                    battLifetime = new lifetime_nmc_t(DataHVACGlobals::TimeStepSys);
+                } else {
+                    // This sets a lifetime model where the capacity is always 100%.
+                    std::vector<double> tblVals{{20, 0, 100, 20, 5000, 100, 20, 10000, 100, 80, 0, 100, 80, 1000, 100, 80, 2000, 100}};
+                    util::matrix_t<double> battLifetimeMatrix(6, 3, &tblVals);
+                    battLifetime = new lifetime_calendar_cycle_t(battLifetimeMatrix, DataHVACGlobals::TimeStepSys);
+                }
+                std::vector<double> batt_losses{0};  // using double because SSC expects a double
+
+                // Create the SSC battery object
+                ssc_battery_ = std::unique_ptr<battery_t>(
```

Comment:
Using the `battery_t` class from SAM Simulation Core (SSC) for the actual battery modeling.